doc/ModulePriority.md

```diff
@@ -0,0 +1,146 @@
+
+Customers who have an array of modules of which some are 'critical' or 'foundation' modules that are required by other modules in the ecosystem might want these modules to be started before other modules. This is so as to achieve a better end user experience where other modules don't have to wait for these 'critical' or 'foundation' modules to be started, so as to initialize themselves.
+
+As an example, some customers want the Edge Hub module to be started before any other non-system modules in the ecosystem are started. This is so that other modules don't spend unnecessary cycles waiting for Edge Hub to come up before they can start sending messages to other modules or upstream to IoT Hub.
+
+**That being said, module owners should design their modules to withstand any failures of these 'critical' or 'foundation' modules, that they are dependent upon, as they could go down at any arbitrary time and an arbitrary number of times.**
+
+## __Configuration__
+
+Customers can optionally specify a `priority` value for each module in their IoT Edge deployment to achieve module boot ordering. Modules with a higher priority will be created and started first and only when an attempt has been made to start them, will other lower priority modules be created and started.
+
+The value of `priority` will be positive and zero-based with 0 being the highest priority value.
+A higher numeric `priority` value would indicate a lower priority assigned to that module. The maximum value of this property will be 4294967295.
+Modules that possess the same priority will be created at the same time and will have no deterministic startup order imposed amongst themselves. Modules with higher priority values will be created and started first and once Edge Agent has made an attempt to create and start them (only the modules with a desired state of `Running` will be started), it will proceed with the creation (and optionally startup) of other lower priority modules in order.
+
+**It must be noted that the Edge Agent module will not support the `priority` property and will by default have the highest priority. As is the behavior currently, Edge Agent will always be the first module to get started.**
+
+Modules that do not have a `priority` value specified will be started in a non-deterministic order and will be assigned a priority of 4294967295 which indicates the lowest possible priority. Edge Agent will trigger the creation of these modules after all modules with a higher priority have been created.
+
+**It must be noted that module boot ordering will initially not be supported for the Kubernetes execution mode of IoT Edge.**
+
+## __Example__
+
+### __How to set priorities of Edge modules__
+
+Here's an example of how to set the priorities of IoT Edge modules through Az CLI:
+
+Create a deployment manifest `deployment.json` JSON file that has your IoT Edge deployment specification. More information about the IoT Edge deployment manifest can be found [here][1].
```

Comment:
As a rule of thumb, "here" is discouraged as link alt-text. For accessibility reasons, it's hard for people with visual impairment to use because customers who use screen readers often scan link alt-text without context of the surrounding sentences. For non-visually impaired readers, it doesn't set good expectations for what they'll see on the other end of that link. For writers, its difficult to maintain especially in the case when a link might break.

Recommended practice is to use the title of the target page.

Reply:
Ah I see. Sure, I'll go ahead and update this.